Add high level diagrams (#26)
kyma-incubator/compass: docs/architecture/high-level-components.md

# Overview

This document describes communication between Management Plane, Application and Runtimes. 

## Components
Management Plane is a central place which stores configuration about Applications and Runtimes and is responsible for propagating that information to Applications and Runtimes. 
Management Plane plays crucial role in establishing trusted connection between Applications and Runtimes. 

> ![](./assets/components-high-level.svg)

Reviewer: Hmm, I think this diagram is a bit too complicated on the left side: Application <-> Management Plane relation. I would simplify it in the following way:

- Remove webhook from this diagram, as IMHO it should be as low level of abstraction as possible
- Remove "add runtime, add application, configure" verbs from Admin and replace it with something like "manage applications and runtimes", or even simpler "manage"
- Does app really configure Management Plane? That's what I read it from the diagram. The arrow should be on the opposite side, right? Because Management Plane passes configuration to Application and Management Plane set-ups trust between App and Runtime. 
- Sync between Agent and Management Plane is two way, right? So the arrows should be on both sizes of the line
-

> ![](./assets/components-high-level.svg)

Author: Admin got fat, are you happy now?

An administrator adds Runtimes, Applications and configures them through Management Plane GraphQL API. 
To keep Runtimes in-sync with the configuration, an agent installed on a Runtime is responsible for fetching actual configuration. 
In case an Application has configured optional Webhooks, Management Plane notify an Application about any events that concern given Application.
Management Plane does not participate in any business flow. After establishing a trusted connection between Application
and Runtime, they communicate directly, with omitting Management Plane. 

## Grouping
In order to connect Applications and Runtimes, assign it to the same Group.
A Group is a simple label with the key `group`. If Application or Runtime is not explicitly assigned to any Group, it belongs to the `default` Group.

> If you add a Group to the Application or Runtime assigned to the `default` group, it is automatically removed from the `default` group and moved to the new one.

Reviewer: Please add an empty line before and after every diagram

> ![](./assets/grouping.svg)

Reviewer: BTW I don't think we can use rounded rectangles here. BTW do we need `LABEL:` in every rectangle? The diagram describes labelling, so I think we can remove it.

> ![](./assets/grouping.svg)

Reviewer: Hmm, Maybe it would be good to include Runtime3, which has label `group=default`. It could be not connected with any app, but the diagram will show the concept with default group.

# Overview

> This document describes communication between Management Plane, Application and Runtimes. 

Reviewer 2: ```suggestion
This document describes communication between Management Plane, Applications and Runtimes. 
```

## Components
Management Plane is a central place which stores configuration about Applications and Runtimes and is responsible for propagating that information to Applications and Runtimes. 
Management Plane plays crucial role in establishing trusted connection between Applications and Runtimes. 

![](./assets/components-high-level.svg)

An administrator adds Runtimes, Applications and configures them through Management Plane GraphQL API. 
To keep Runtimes in-sync with the configuration, an agent installed on a Runtime is responsible for fetching actual configuration. 

> An Agent sends information about Runtime healthchecks to Management Plane. 

Reviewer 2: ```suggestion
An Agent sends information about Runtime health checks to Management Plane. 
```

> In case an Application has configured optional Webhooks, Management Plane notify an Application about any events that concern given Application.

Reviewer 2: ```suggestion
In case an Application has configured optional Webhooks, Management Plane notifies an Application about any events that concern given Application.
```

Management Plane does not participate in any business flow. After establishing a trusted connection between Application
and Runtime, they communicate directly, with omitting Management Plane. 

## Grouping
In order to connect Applications and Runtimes, assign it to the same Group.
A Group is a simple label with the key `group`. If Application or Runtime is not explicitly assigned to any Group, it belongs to the `default` Group.
If you add a Group to the Application or Runtime assigned to the `default` group, it is automatically removed from the `default` group and moved to the new one.

> ![](./assets/grouping.svg)

Reviewer 2: After the changes to this diagram, the red arrow still seems a bit confusing to me, because, for example, `App1` can't communicate with `Runtime2` or `Runtime3` either and yet there's no red arrow connecting them. To understand what's happening reader has to read the whole next paragraph. Maybe that's just me, though.

Reviewer 3: I agree, for me, this red arrow is also a bit confusing.

> # High Level Components

Reviewer 4: Please add an entry (as a first one) in /docs/README.md inside architecture section
